runners/core-construction-java/src/test/java/org/apache/beam/runners/core/construction/WindowingStrategyTranslationTest.java

```diff
@@ -84,7 +90,18 @@ private static ToProtoAndBackSpec toProtoAndBackSpec(WindowingStrategy windowing
                 .withMode(AccumulationMode.DISCARDING_FIRED_PANES)
                 .withTrigger(REPRESENTATIVE_TRIGGER)
                 .withAllowedLateness(Duration.millis(93))
-                .withTimestampCombiner(TimestampCombiner.LATEST)));
+                .withTimestampCombiner(TimestampCombiner.LATEST)),
+        toProtoAndBackSpec(
+            WindowingStrategy.of(
+                RawWindowFn.forFunctionSpec(
+                    SdkFunctionSpec.newBuilder()
+                        .setEnvironmentId("env")
+                        .setSpec(
+                            FunctionSpec.newBuilder()
+                                .setUrn("env_custom_window_fn")
+                                .setPayload(ByteString.EMPTY)
```

Comment:
A non-empty `ByteString` would be more meaningful as a test value, yes?

Reply:
Done.